app/parse.py

```diff
@@ -15,9 +26,266 @@ class Product:
+                (By.CSS_SELECTOR, "#closeCookieBanner")
+            )  # .acceptCookies
+        )
+        cookie_button.click()
+
+    except (NoSuchElementException, ElementClickInterceptedException):
+        pass
+
+
+def get_all_products_from_page(
+    url: str, driver: WebDriver, main_selector: str, selector: str
+) -> list[Product]:
+    products = []
+    driver.get(url)
+
+    products_link = driver.find_element(By.CSS_SELECTOR, main_selector)
+    products_link.click()
+
+    WebDriverWait(driver, 10).until(
+        expected_conditions.presence_of_element_located((
+            By.CSS_SELECTOR,
+            ".nav.nav-second-level"
+        ))
+    )
+
+    all_products_link = driver.find_element(By.CSS_SELECTOR, selector)
+    all_products_link.click()
+
+    WebDriverWait(driver, 10).until(
+        expected_conditions.presence_of_element_located(
+            (By.CSS_SELECTOR, ".row.ecomerce-items.ecomerce-items-more")
+        )
+    )
+
+    count_more_btn = 1
+    while True:
+        try:
+            more_btn = driver.find_element(
+                By.CSS_SELECTOR,
+                ".btn.btn-lg.btn-block.btn-primary.ecomerce-items-scroll-more",
+            )
+            if more_btn:
+                more_btn.click()
+            else:
+                break
+
+            WebDriverWait(driver, 10).until(
+                expected_conditions.presence_of_element_located((
+                    By.CSS_SELECTOR,
+                    ".row.ecomerce-items.ecomerce-items-more"
+                ))
+            )
+            print("CLICK MORE BTN ", count_more_btn)
+            count_more_btn += 1
+        except ElementClickInterceptedException:
+            close_cookie_banner(driver)
+        except ElementNotInteractableException:
+            print("exit from loop")
+            break
+
+    print("we are waiting")
+    WebDriverWait(driver, 10).until(
+        expected_conditions.presence_of_element_located((
+            By.CSS_SELECTOR, ".col-md-4.col-xl-4.col-lg-4"
+        ))
+    )
+    print("we are taking all cards")
+
+    cards = driver.find_elements(
+        By.CSS_SELECTOR, ".col-md-4.col-xl-4.col-lg-4"
+    )
+
+    print("we will start a loop")
+
+    for card in cards:
+        products.append(
+            Product(
+                title=card.find_element(
+                    By.CLASS_NAME,
+                    "title"
+                ).get_attribute("title"),
+                description=card.find_element(
+                    By.CLASS_NAME,
+                    "description"
+                ).text,
+                price=float(
+                    card.find_element(
+                        By.CLASS_NAME,
+                        "price"
+                    ).text.replace("$", "")
+                ),
+                rating=len(
+                    card.find_elements(
+                        By.CSS_SELECTOR,
+                        ".ratings p span.ws-icon"
+                    )
+                ),
```

Comment:
The `rating` field is being calculated using the length of elements found with the selector `.ratings p span.ws-icon`. This may not correctly represent the actual rating value. Ensure that the correct attribute or text is being used to extract the rating.